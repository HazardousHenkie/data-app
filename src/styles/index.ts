import { createGlobalStyle } from 'styled-components'

const GlobalStyle = createGlobalStyle`
  html,
  body,
  #root {
    height: 100%;
  }

  .MuiPopover-root {
    z-index: 1400 !important;
  }
`

export default GlobalStyle
