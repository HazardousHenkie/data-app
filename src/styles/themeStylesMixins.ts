const MuiDrawer = {
    padding: '20px',
    borderTopRightRadius: '25px',
    borderTopLeftRadius: '25px'
}

export default MuiDrawer
