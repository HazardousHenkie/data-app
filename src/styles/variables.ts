const variables = {
    black: '#000',
    white: '#fff',
    gray: '#D3D3D3',
    darkGray: '#A9A9A9',
    breakpoints: {
        up: {
            xxs: 'min-width: 0',
            xs: 'min-width: 420px',
            sm: 'min-width: 600px',
            md: 'min-width: 960px',
            lg: 'min-width: 1280px',
            xl: 'min-width: 1920px'
        },
        down: {
            xs: 'max-width: 420px',
            sm: 'max-width: 599px',
            md: 'max-width: 959px',
            lg: 'max-width: 1279px',
            xl: 'max-width: 1919px'
        }
    }
}

export default variables
