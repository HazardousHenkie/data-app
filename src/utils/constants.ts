enum ENDPOINTS {
    LOGOUT = '/.netlify/functions/googleLogout',
    lOGIN = '/.netlify/functions/googleLogin',
    REFRESH_TOKEN = '/.netlify/functions/refreshToken',
    COUNTRIES = '/.netlify/functions/countries',
    FAVORITE_COUNTRIES = '/.netlify/functions/getFavoritedCountries',
    SAVE_COUNTRY = '/.netlify/functions/saveCountry/',
    DELETE_COUNTRY = '/.netlify/functions/deleteCountry/'
}

export default ENDPOINTS
