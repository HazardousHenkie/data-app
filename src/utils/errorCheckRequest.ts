import { put } from 'redux-saga/effects'
import { getRefreshTokenRequest } from 'globals/authentication/refreshToken/actions'
import { logoutRequest } from 'globals/authentication/logout/actions'

import { setError } from 'globals/globalErrors/actions'
import { ResponseError } from './request'

// test this one?
function* requestErrorCheck(error: ResponseError) {
    try {
        if (localStorage.getItem('userId')) {
            yield put(
                getRefreshTokenRequest(localStorage.getItem('userId') as string)
            )
        } else {
            throw Error('No userId in local storage.')
        }
    } catch {
        yield put(logoutRequest())
        yield put(setError(error))
    }
}
export default requestErrorCheck
