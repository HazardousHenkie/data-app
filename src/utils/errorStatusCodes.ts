enum ERROR_STATUS_CODES {
    BAD_REQUEST = 400,
    UNAUTHORIZED = 401,
    SERVER_ERROR = 500,
    SERVICE_UNAVAILABLE = 503,
    NOT_FOUND = 404
}

export default ERROR_STATUS_CODES
