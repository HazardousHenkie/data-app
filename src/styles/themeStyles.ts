import { createMuiTheme, responsiveFontSizes } from '@material-ui/core/styles'

let lightTheme = createMuiTheme({
  palette: {
    type: 'light',
    primary: {
      main: '#343434'
    }
  }
})

export let darkTheme = responsiveFontSizes(
  createMuiTheme({
    palette: {
      type: 'dark',
      primary: {
        main: '#343434'
      }
    }
  })
)

export default lightTheme = responsiveFontSizes(lightTheme)
