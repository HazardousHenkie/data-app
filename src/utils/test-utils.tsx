import React from 'react'
import { render } from '@testing-library/react'
import configureStore from 'configureStore'

import history from 'utils/history'
import { Provider } from 'react-redux'

import variables from 'styles/variables'
import { ThemeProvider } from 'styled-components'
import { ThemeProvider as MuiThemeProvider } from '@material-ui/styles'
import lightTheme from 'styles/themeStyles'

const customRender = (
    ui: React.ReactElement,
    {
        initialState = {},
        store = configureStore(initialState, history),
        ...renderOptions
    } = {}
) => {
    const Wrapper: React.FC = ({ children }) => {
        return (
            <Provider store={store}>
                <ThemeProvider theme={variables}>
                    <MuiThemeProvider theme={lightTheme}>
                        <ThemeProvider theme={lightTheme}>
                            {children}
                        </ThemeProvider>
                    </MuiThemeProvider>
                </ThemeProvider>
            </Provider>
        )
    }
    return render(ui, {
        wrapper: Wrapper,
        ...renderOptions
    })
}

export * from '@testing-library/react'

export { customRender as render }
