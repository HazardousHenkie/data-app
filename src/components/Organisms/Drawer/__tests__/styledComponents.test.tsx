import React from 'react'

import { render } from 'utils/test-utils'

import { css } from 'styled-components'

import { Theme } from '@material-ui/core/styles'
import lightTheme, { darkTheme } from 'styles/themeStyles'

import {
    SwipeableDrawerStyled,
    HandleBar,
    DrawerWrapper,
    ClickIndicator,
    SwipeIndicator,
    swipeAnimation,
    SwipeIndicatorInner,
    SwipeIndicatorInnerDown
} from '../styledComponents'

import 'jest-styled-components'

describe('<SwipeableDrawerStyled />', () => {
    const renderWithTheme = (theme?: Theme) =>
        render(
            <SwipeableDrawerStyled
                theme={theme || lightTheme}
                anchor="bottom"
                open
                onClose={() => {}}
                onOpen={() => {}}
            />
        )

    it('should render and match the snapshot', () => {
        const { container } = renderWithTheme()

        expect(container.firstChild).toMatchSnapshot()
    })

    it('should render a <DIV> tag', () => {
        const { container } = renderWithTheme()

        expect(container.firstElementChild?.tagName).toEqual('DIV')
    })
})

describe('<HandleBar />', () => {
    const renderWithThemeHandleBar = (theme?: Theme) =>
        render(<HandleBar theme={theme || lightTheme} />)

    it('should render and match the snapshot', () => {
        const { container } = renderWithThemeHandleBar()

        expect(container.firstChild).toMatchSnapshot()
    })

    it('should render a <DIV> tag', () => {
        const { container } = renderWithThemeHandleBar()

        expect(container.firstElementChild?.tagName).toEqual('DIV')
    })

    it('should have position relative if relative props is passed', () => {
        const { container } = render(<HandleBar relative />)

        expect(container.firstChild).toHaveStyle('position: relative')
    })

    it('should have position relative if relative props is passed', () => {
        const { container } = render(<HandleBar />)

        expect(container.firstChild).toHaveStyle('position: fixed')
    })

    it('should render breakpoint and display none', () => {
        const { container } = renderWithThemeHandleBar()

        expect(container.firstChild).toHaveStyleRule('display', 'none', {
            media: `${lightTheme.breakpoints.up('md').slice(7)}`
        })
    })

    it('should render light variable when LightTheme is set', () => {
        const { container } = renderWithThemeHandleBar()

        expect(container.firstChild).toHaveStyle(
            `background: ${lightTheme.palette.common.black}`
        )

        expect(container.firstChild).toHaveStyle(
            ` border: 2px solid ${lightTheme.palette.common.black}`
        )
    })

    it('should render light variable when darkmode is set', () => {
        const { container } = renderWithThemeHandleBar(darkTheme)

        expect(container.firstChild).toHaveStyle(
            `background: ${darkTheme.palette.common.black}`
        )

        expect(container.firstChild).toHaveStyle(
            ` border: 2px solid ${darkTheme.palette.common.black}`
        )
    })
})

describe('<DrawerWrapper />', () => {
    it('should render and match the snapshot', () => {
        const { container } = render(<DrawerWrapper />)

        expect(container.firstChild).toMatchSnapshot()
    })

    it('should render a <DIV> tag', () => {
        const { container } = render(<DrawerWrapper />)

        expect(container.firstElementChild?.tagName).toEqual('DIV')
    })
})

describe('<ClickIndicator />', () => {
    const renderWithClickIndicator = (theme?: Theme) =>
        render(<ClickIndicator theme={theme || lightTheme} />)

    it('should render and match the snapshot', () => {
        const { container } = renderWithClickIndicator()

        expect(container.firstChild).toMatchSnapshot()
    })

    it('should render a <DIV> tag', () => {
        const { container } = renderWithClickIndicator()

        expect(container.firstElementChild?.tagName).toEqual('DIV')
    })

    it('should render breakpoint and display none', () => {
        const { container } = renderWithClickIndicator()

        expect(container.firstChild).toHaveStyleRule('display', 'none', {
            media: `${lightTheme.breakpoints.down('sm').slice(7)}`
        })
    })

    it('should render indicator with lightTheme', () => {
        const { container } = renderWithClickIndicator()

        expect(container.firstChild).toHaveStyle(
            `border: 2px solid ${lightTheme.palette.common.black}`
        )
    })

    it('should render indicator with darkTheme', () => {
        const { container } = renderWithClickIndicator(darkTheme)

        expect(container.firstChild).toHaveStyle(
            `border: 2px solid ${darkTheme.palette.common.black}`
        )
    })
})

describe('<SwipeIndicator />', () => {
    const renderWithSwipeIndicator = (theme?: Theme) =>
        render(<SwipeIndicator theme={theme || lightTheme} />)

    it('should render and match the snapshot', () => {
        const { container } = renderWithSwipeIndicator()

        expect(container.firstChild).toMatchSnapshot()
    })

    it('should render a <DIV> tag', () => {
        const { container } = renderWithSwipeIndicator()

        expect(container.firstElementChild?.tagName).toEqual('DIV')
    })

    it("should render bottom style when down prop isn't given", () => {
        const { container } = renderWithSwipeIndicator()

        expect(container.firstChild).toHaveStyle('bottom: 10px')
    })

    it('should render top style when down prop is given', () => {
        const { container } = render(<SwipeIndicator down />)

        expect(container.firstChild).toHaveStyle('top: -35px')
    })

    it('should render indicator with lightTheme', () => {
        const { container } = renderWithSwipeIndicator()

        expect(container.firstChild).toHaveStyle(
            `border: 2px solid ${lightTheme.palette.common.black}`
        )
    })

    it('should render indicator with darkTheme', () => {
        const { container } = renderWithSwipeIndicator(darkTheme)

        expect(container.firstChild).toHaveStyle(
            `border: 2px solid ${darkTheme.palette.common.black}`
        )
    })
})

describe('<SwipeIndicatorInner />', () => {
    const renderWithSwipeIndicatorInner = (theme?: Theme) =>
        render(<SwipeIndicatorInner theme={theme || lightTheme} />)

    it('should render and match the snapshot', () => {
        const { container } = renderWithSwipeIndicatorInner()

        expect(container.firstChild).toMatchSnapshot()
    })

    it('should render a <SVG> tag', () => {
        const { container } = renderWithSwipeIndicatorInner()

        expect(container.querySelector('svg')).toBeInTheDocument()
    })

    it('should render animation style', () => {
        const { container } = renderWithSwipeIndicatorInner()

        expect(container.firstChild).toHaveStyleRule(
            (css`
                animation: ${swipeAnimation(false)} 2s infinite;
            ` as unknown) as string
        )
    })

    it('should render indicator with lightTheme', () => {
        const { container } = renderWithSwipeIndicatorInner()

        expect(container.firstChild).toHaveStyle(
            `color: ${lightTheme.palette.common.black}`
        )
    })

    it('should render indicator with darkTheme', () => {
        const { container } = renderWithSwipeIndicatorInner(darkTheme)

        expect(container.firstChild).toHaveStyle(
            `color: ${darkTheme.palette.common.black}`
        )
    })
})

describe('<SwipeIndicatorInnerDown />', () => {
    const renderWithSwipeIndicatorInnerDown = (theme?: Theme) =>
        render(<SwipeIndicatorInnerDown theme={theme || lightTheme} />)

    it('should render and match the snapshot', () => {
        const { container } = renderWithSwipeIndicatorInnerDown()

        expect(container.firstChild).toMatchSnapshot()
    })

    it('should render a <SVG> tag', () => {
        const { container } = renderWithSwipeIndicatorInnerDown()

        expect(container.querySelector('svg')).toBeInTheDocument()
    })

    it('should render animation style', () => {
        const { container } = renderWithSwipeIndicatorInnerDown()

        expect(container.firstChild).toHaveStyleRule(
            (css`
                animation: ${swipeAnimation(true)} 2s infinite;
            ` as unknown) as string
        )
    })

    it('should render indicator with lightTheme', () => {
        const { container } = renderWithSwipeIndicatorInnerDown()

        expect(container.firstChild).toHaveStyle(
            `color: ${lightTheme.palette.common.black}`
        )
    })

    it('should render indicator with darkTheme', () => {
        const { container } = renderWithSwipeIndicatorInnerDown(darkTheme)

        expect(container.firstChild).toHaveStyle(
            `color: ${darkTheme.palette.common.black}`
        )
    })
})
