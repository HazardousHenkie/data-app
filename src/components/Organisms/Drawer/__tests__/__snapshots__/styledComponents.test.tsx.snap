// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClickIndicator /> should render and match the snapshot 1`] = `
.c0 {
  cursor: pointer;
  opacity: 0.5;
  position: absolute;
  left: 50%;
  width: 30px;
  height: 30px;
  margin-left: -15px;
  border: 2px solid #000;
  border-radius: 50px;
  bottom: 10px;
  z-index: 10;
}

@media (max-width:959.95px) {
  .c0 {
    display: none;
  }
}

<div
  class="c0"
/>
`;

exports[`<DrawerWrapper /> should render and match the snapshot 1`] = `
.c0 {
  position: relative;
  bottom: 0;
}

<div
  class="c0"
/>
`;

exports[`<HandleBar /> should render and match the snapshot 1`] = `
.c0 {
  border-radius: 100px;
  width: 50%;
  bottom: 0;
  top: auto;
  position: fixed;
  left: 25%;
  margin: 0 auto 8px;
  background: #000;
  border: 2px solid #000;
}

@media (min-width:960px) {
  .c0 {
    display: none;
  }
}

<div
  class="c0"
/>
`;

exports[`<SwipeIndicator /> should render and match the snapshot 1`] = `
.c0 {
  cursor: pointer;
  opacity: 0.5;
  position: absolute;
  left: 50%;
  width: 30px;
  height: 30px;
  margin-left: -15px;
  border: 2px solid #000;
  border-radius: 50px;
  bottom: 10px;
}

<div
  class="c0"
/>
`;

exports[`<SwipeIndicatorInner /> should render and match the snapshot 1`] = `
.c0 {
  position: absolute;
  left: 50%;
  margin-left: -12px;
  color: #000;
  top: 5px;
  -webkit-animation: kQWVHZ 2s infinite;
  animation: kQWVHZ 2s infinite;
}

<svg
  aria-hidden="true"
  class="MuiSvgIcon-root c0"
  focusable="false"
  theme="[object Object]"
  viewBox="0 0 24 24"
>
  <path
    d="M7 14l5-5 5 5z"
  />
</svg>
`;

exports[`<SwipeIndicatorInnerDown /> should render and match the snapshot 1`] = `
.c0 {
  position: absolute;
  left: 50%;
  margin-left: -12px;
  color: #000;
  top: -5px;
  -webkit-animation: faQfbw 2s infinite;
  animation: faQfbw 2s infinite;
}

<svg
  aria-hidden="true"
  class="MuiSvgIcon-root c0"
  focusable="false"
  theme="[object Object]"
  viewBox="0 0 24 24"
>
  <path
    d="M7 10l5 5 5-5z"
  />
</svg>
`;

exports[`<SwipeableDrawerStyled /> should render and match the snapshot 1`] = `
<div
  class="PrivateSwipeArea-root-1 PrivateSwipeArea-anchorBottom-5"
  style="height: 20px;"
/>
`;
